### PR TITLE
[tfa-fix] Add fix for keyring not present error

### PR DESCRIPTION
**Pull request by vinpapnoi**
# Description

Problem:

Test suite `tier-1_cephadm_operations_via_admin_node` failed to validate keyring is present in node after adding label `_admin` to the node.

Reason for Failure:

There was no waiter added after label `_admin` was added to the node and the test was trying to search for keyring before it was updated.

Solution:

Added a waiter after label addition for the keyring files validation to give the system time to udpate the keyrings. Also added waiter for `ceph orch ls` validation.

### CODE REVIEW

**Review by tintumathew10 — COMMENTED** [review]
lgtm

File: tests/cephadm/test_cephadm_operations_via_admin.py [hunk 1]
```diff
@@ -44,9 +45,12 @@ def run(ceph_cluster, **kw):
...
     # Verify files in new admin node
-    for file in files:
-        result = os.path.exists(file)
-        if result is False:
+    timeout, interval = 300, 6
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        for file in files:
+            if os.path.exists(file):
+                break
+        if w.expired:
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_operations_via_admin.py:53]: This check should be outside of waiter loop

File: tests/cephadm/test_cephadm_operations_via_admin.py [hunk 2]
```diff
@@ -44,9 +45,12 @@ def run(ceph_cluster, **kw):
             f"Failed to add label '{label}' on node '{hostname}'"
         )
     # Verify files in new admin node
-    for file in files:
-        result = os.path.exists(file)
-        if result is False:
+    timeout, interval = 300, 6
+    for w in WaitUntil(timeout=timeout, interval=interval):
+        for file in files:
+            if os.path.exists(file):
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_operations_via_admin.py:51]: Check is for all files ? this will break the loop even single file from `files` is present.

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: tests/cephadm/test_cephadm_operations_via_admin.py (hunk 2 above)
Comment (vinpapnoi) [reply tests/cephadm/test_cephadm_operations_via_admin.py]: Changed to check for all files (two)

File: tests/cephadm/test_cephadm_operations_via_admin.py (hunk 1 above)
Comment (vinpapnoi) [reply tests/cephadm/test_cephadm_operations_via_admin.py]: Done

**Comment by vinpapnoi** [comment]
Updated test run log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-O2OQ8J

File: tests/cephadm/test_cephadm_operations_via_admin.py [hunk 3]
```diff
@@ -44,14 +45,26 @@ def run(ceph_cluster, **kw):
...
         )
     # Verify files in new admin node
-    for file in files:
-        result = os.path.exists(file)
-        if result is False:
-            raise CephadmOperationsAdmin(
-                f"File '{file}' not present on node '{hostname}'"
-            )
+    timeout, interval = 300, 6
+    for w in WaitUntil(timeout=timeout, interval=interval):
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_operations_via_admin.py:49]: Use below optimised code -
```
  for w in WaitUntil(timeout=timeout, interval=interval):
    _files = []
    for _f in files:
      if os.path.exists(file):
        _files.append(_f)
    
    if len(_files) == len(files):
      break
```

Comment (vinpapnoi) [reply tests/cephadm/test_cephadm_operations_via_admin.py:49]: Done

**Review by tintumathew10 — DISMISSED** [review]
Changes looks good

**Review by MohitBis — DISMISSED** [review]
LGTM

File: tests/cephadm/test_cephadm_operations_via_admin.py [hunk 4]
```diff
@@ -44,14 +45,23 @@ def run(ceph_cluster, **kw):
...
+        _files = []
```
Comment (vamahaja) [inline tests/cephadm/test_cephadm_operations_via_admin.py:50]: pythonic way -
```
_files = [_f for _f in files if os.path.exists(_f)]
```

Comment (vinpapnoi) [reply tests/cephadm/test_cephadm_operations_via_admin.py]: Done

**Comment by vinpapnoi** [comment]
Updated test run logs: http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-LDJ01O/

**Review by vamahaja — APPROVED** [review]

**Pull request merged**